sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -831,7 +832,14 @@ case class HashAggregateExec(
     ctx.currentVars = new Array[ExprCode](aggregateBufferAttributes.length) ++ input
 
     val updateRowInRegularHashMap: String = {
-      ctx.INPUT_ROW = unsafeRowBuffer
+      val updatedTmpAggBuffer =
+        if (isFastHashMapEnabled && !isVectorizedHashMapEnabled) {
+          updatedAggBuffer
```

Comment:
just realized it. Do we create the `updatedAggBuffer` variable only to improve the readability of the generated code? It looks to me we don't need this variable. Here we can write
```
ctx.INPUT_ROW = if (isFastHashMapEnabled && !isVectorizedHashMapEnabled) fastRowBuffer else unsafeRowBuffer
```

Reply:
@cloud-fan , I will update it. thanks.

Reply:
Did you update? This is not what I proposed...